libproc/proc.c

```diff
@@ -99,10 +106,27 @@ _open (pctx_t ctx)
 int
 proc_open (pctx_t ctx, const char *path)
 {
+    struct stat buf;
+    int rc;
+
     assert (ctx->pctx_magic == PCTX_MAGIC);
     assert (!ctx->pctx_fp && !ctx->pctx_dp);
 
+    snprintf (ctx->pctx_root, sizeof (ctx->pctx_root), "%s%s/",
+      ctx->pctx_real_root, PROC_ROOT_SYS);
+    ctx->pctx_path = ctx->pctx_root + strlen (ctx->pctx_root);
+    ctx->pctx_pathlen = sizeof (ctx->pctx_root) - strlen (ctx->pctx_root);
     snprintf (ctx->pctx_path, ctx->pctx_pathlen, "%s", path);
+
+    if ((rc = stat(ctx->pctx_root, &buf)) != 0) {
+        snprintf (ctx->pctx_root, sizeof (ctx->pctx_root), "%s%s/",
+          ctx->pctx_real_root, PROC_ROOT_PROC);
+        ctx->pctx_path = ctx->pctx_root + strlen (ctx->pctx_root);
+        ctx->pctx_pathlen = sizeof (ctx->pctx_root) - strlen (ctx->pctx_root);
+        snprintf (ctx->pctx_path, ctx->pctx_pathlen, "%s", path);
+        rc = stat(ctx->pctx_root, &buf);
```

Comment:
This stat() probably isn't necessary either since it has no effect.

I'm not sure what we would do at this point in the code either if both /proc and /sys were missing.  Having /proc be the second choice likely makes it behave like it would in the past if /proc disappeared, so there is probably no need for additional test here.

Reply:
I agree.